app/src/sharedTest/java/org/oppia/app/utility/DragViewAction.kt

```diff
@@ -32,98 +33,116 @@ private const val DRAG_DURATION = 1500
-        // See android.test.TouchUtils longClickView
-        val longPressTimeout = (ViewConfiguration.getLongPressTimeout() * 1.5f).toLong()
-        uiController.loopMainThreadForAtLeast(longPressTimeout)
-        return downEvent
       }
 
-      override fun sendSwipe(uiController: UiController, startCoordinates: FloatArray, endCoordinates: FloatArray, precision: FloatArray): Swiper.Status {
-        return sendLinearDrag(uiController, downMotion, startCoordinates, endCoordinates, precision)
-      }
+    private fun performLongPress(
+      uiController: UiController,
+      coordinates: FloatArray,
+      precision: FloatArray
+    ): MotionEvent {
+      val downEvent = MotionEvents.sendDown(uiController, coordinates, precision).down
+      // Duration before a press turns into a long press.
+      // Factor 1.5 is needed, otherwise a long press is not safely detected.
+      // See android.test.TouchUtils longClickView
+      val longPressTimeout = (ViewConfiguration.getLongPressTimeout() * 1.5f).toLong()
+      uiController.loopMainThreadForAtLeast(longPressTimeout)
+      return downEvent
+    }
 
-      private fun sendLinearDrag(uiController: UiController, downMotion: DragViewAction.DownMotionPerformer, startCoordinates: FloatArray, endCoordinates: FloatArray, precision: FloatArray): Swiper.Status {
-        val steps = interpolate(startCoordinates,endCoordinates)
-        val delayBetweenMovements = DRAG_DURATION / steps.size
-        val downEvent = downMotion.perform(uiController, startCoordinates, precision) ?: return Swiper.Status.FAILURE
-        try {
-          steps.forEachIndexed { index, step ->
-            if (!MotionEvents.sendMovement(uiController, downEvent, step)) {
-              MotionEvents.sendCancel(uiController, downEvent)
-              return Swiper.Status.FAILURE
-            }
-            val desiredTime: Long = downEvent.downTime + delayBetweenMovements * index
-            val timeUntilDesired = desiredTime - SystemClock.uptimeMillis()
-            if (timeUntilDesired > 10) {
-              // If the wait time until the next event isn't long enough, skip the wait
-              // and execute the next event.
-              uiController.loopMainThreadForAtLeast(timeUntilDesired)
-            }
-          }
-          // Wait before sending up because some drag handling logic may discard move events
-          // that has been sent immediately before the up event. e.g. HandleView.
-          uiController.loopMainThreadForAtLeast(WAIT_BEFORE_SENDING_UP.toLong())
-          if (!MotionEvents.sendUp(uiController, downEvent, endCoordinates)) {
+    override fun sendSwipe(
+      uiController: UiController,
+      startCoordinates: FloatArray,
+      endCoordinates: FloatArray,
+      precision: FloatArray
+    ): Swiper.Status {
+      return sendLinearDrag(uiController, downMotion, startCoordinates, endCoordinates, precision)
+    }
+
+    private fun sendLinearDrag(
```

Comment:
We shouldn't be copying code from Espresso--is there some way to use existing functionality from Espresso to do perform this functionality, or another library that we can integrate to do it for us? Otherwise, we really should be writing it from scratch rather than copying or rewriting Espresso's implementation.

Reply:
Actually I couldn't find any usage of this that's why i did create this in the first place i asked Rajat as well for the same he couldn't find anything else.

Reply:
That means we need to write this from scratch. It's fine to use the existing Espresso classes as a reference, but we shouldn't be copying code from it.